cluster/charts/crossplane/templates/service.yaml

```diff
@@ -20,6 +20,6 @@ spec:
     release: {{ .Release.Name }}
   ports:
   - protocol: TCP
-    port: 9443
-    targetPort: 9443
+    port: {{ .Values.webhooks.port | default "9443" }}
```

Comment:
You're right @ravilr, I didn't think about this before, but the service port can remain the same. I will refactor this again, but I need your input on this, as the override makes it a bit awkward:

- In the runtime config the user can provide a service port with a port diverging from this
- The runtime config just embeds the upstream corev1.ServiceSpec so we can't really prevent this from happening

Would you prefer 
1. the `runtime_defaults.go#serviceFromRuntimeConfig` method to override it if present, 
2. a new default override `ServiceWithPort(serviceName string, port int)` to be added to the `allOverrides` method

I'm unsure which one is actually less surprising when reading the code. I think I'm leaning towards the second one

Reply:
1. In case of crossplane core cli, the newly being added `--webhook-port` arg  appears to be unambiguous already, capturing the webhook server listener port in the crossplane pod's core container. So, i think the crossplane chart can continue hard-coding the webhook `Service` spec port to 9443 as before.

2. In case of pkg DeploymentRuntimeConfig,  2nd option from your proposal sounds good to me, since we are already expecting to the DRC's serviceTemplate.spec override to be matching on the port name `webhook`.

cc @phisco @turkenh for the maintainers to weigh in.

Reply:
The chart now hard-codes 9443 as the service port and an override was added to ensure the port is always 9443. Test case was adjusted accordingly